src/librustc_mir/transform/const_prop.rs

```diff
@@ -1004,11 +1004,27 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                     let expected = ScalarMaybeUninit::from(Scalar::from_bool(*expected));
                     let value_const = self.ecx.read_scalar(value).unwrap();
                     if expected != value_const {
+                        enum DbgVal<T> {
```

Comment:
This looks like something more generally useful that shouldn't be 5 layers deep inside a private function?

Reply:
it's very specifically tailored to this case. I'm sure we could come up with something more general and put it into `rustc_data_structures`